src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/Exceptions.cs

```diff
@@ -24,8 +27,46 @@ private static string GetExceptionMessage(VmrIngestionPatch patch, ProcessExecut
             + result;
 }
 
-public class ConflictInPrBranchException(VmrIngestionPatch patch, string targetBranch)
+public class ConflictInPrBranchException(ProcessExecutionResult conflictResult, string targetBranch, bool isForwardFlow)
     : Exception($"Failed to flow changes due to conflicts in the target branch ({targetBranch})")
 {
-    public VmrIngestionPatch Patch { get; } = patch;
+    public List<string> FilesInConflict { get; } = ParseResult(conflictResult, isForwardFlow);
+
+    private const string AlreadyExistsRegex = "patch failed: (.+): already exist in index";
+    private const string PatchFailedRegex = "error: patch failed: (.*):";
+    private const string PatchDoesNotApplyRegex = "error: (.+): patch does not apply";
+    private const string FileDoesNotExistRegex = "error: (.+): does not exist in index";
```

Comment:
I think these should be instantiated as `Regex` static readonly instances so that we don't keep compiling the strings to regexes every time we need to parse the errors.